packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -205,12 +221,13 @@ class ChangeNotifier implements Listenable {
   @visibleForTesting
   void notifyListeners() {
     assert(_debugAssertNotDisposed());
-    if (_listeners != null) {
-      final List<VoidCallback> localListeners = List<VoidCallback>.from(_listeners!);
-      for (final VoidCallback listener in localListeners) {
+    _firstNewlyAddedEntry = null;
+    if (_listeners != null && _listeners!.isNotEmpty) {
+      for (_ListenerEntry? entry = _listeners!.first;
```

Comment:
Why's entry nullable?

Reply:
`entry.next` may be `null` when reaching the end of the list. We need it to be nullable for the `entry = entry.next`

Reply:
Makes sense.